notebooks/06_Serving/06_01_Seldon_Core_Serving.ipynb

```diff
@@ -0,0 +1,219 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl label namespace kubeflow serving.kubeflow.org/inferenceservice=enabled"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Create an example SeldonDeployment with a dummy model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl create -n kubeflow -f seldondeployment.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3.Wait for state to become available"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl get sdep seldon-model -n kubeflow -o jsonpath='{.status.state}\\n'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Port forward to the Istio gateway"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl port-forward $(kubectl get pods -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].metadata.name}') -n istio-system 8004:80"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Send a prediction request"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!curl -s -d '{\"data\": {\"ndarray\":[[1.0, 2.0, 5.0]]}}'    -X POST http://localhost:8004/seldon/kubeflow/seldon-model/api/v1.0/predictions    -H \"Content-Type: application/json\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You should be able to see a response like below:\n",
+    "\n",
+    "```\n",
+    "{\n",
+    "  \"meta\": {\n",
+    "    \"puid\": \"i2e1i8nq3lnttadd5i14gtu11j\",\n",
+    "    \"tags\": {\n",
+    "    },\n",
+    "    \"routing\": {\n",
+    "    },\n",
+    "    \"requestPath\": {\n",
+    "      \"classifier\": \"seldonio/mock_classifier_rest:1.3\"\n",
+    "    },\n",
+    "    \"metrics\": []\n",
+    "  },\n",
+    "  \"data\": {\n",
+    "    \"names\": [\"proba\"],\n",
+    "    \"ndarray\": [[0.43782349911420193]]\n",
+    "  }\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6. Canary Rollout"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl apply -n kubeflow -f canary.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Run bash script to send 100 times prediction request"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!chmod +x ./predict_request.sh\n",
```

Comment:
Can we get ride of a separate shell scripts? You do use curl at the beginning. predict_request.sh doesn't have a few lines. why not just move to notebook? 

%bash
your prediction scripts here?

Reply:
Yeah, it makes sense. We don't have to use scripts but use bash in-line.